### PR TITLE
Enhanced push() to return this

### LINKED ISSUE #3: Make push() fluent
This for this useful package. This is an enhancement request to make the push() function fluent. 

example:
```
movingAvg.
  push(100).
  get();
```
I'll submit a PR with this small change for your consideration.

### PR DESCRIPTION
**Pull request by wayneparrott**
Enhanced push() for fluent style use.

* index.js - modifed push()
* readme.md - updated example to illustrate chaining multiple push() with a get() and updated push() doc
* test.js - added test
* example.js - included chaining push() calls example

Fix #3

### CODE REVIEW

File: example.js [hunk 1]
```diff
@@ -5,10 +5,11 @@ const createWindow = require('.')
 const avgOfLast3 = createWindow(3, 0)
 avgOfLast3.get() // 0
 
-avgOfLast3.push(2)
-avgOfLast3.push(3)
-avgOfLast3.push(4)
-avgOfLast3.get() // (2 + 3 + 4) / 3 = 3
+avgOfLast3.
+    push(2).
```
Comment (derhuerst) [inline example.js]: I'd like to have the `.` in the beginning of the line. This makes commenting in/out and line-wise [staging](https://git-scm.com/docs/git-add) easier.

Same with `readme.md`.

File: readme.md [hunk 2]
```diff
@@ -54,7 +56,8 @@ avg.get() // 6, because (5 + 6 + 7) / 3
 
 ### `w.push(val)`
 
-`val` must be a number.
+`val` must be a number.    
+Return `this`.
```
Comment (derhuerst) [inline readme.md]: I think this phrasing is a bit confusing (this is my subjective judgement though). What do you think about "Returns `m`." ?

**Review by derhuerst — COMMENTED** [review]
Thanks for your PR!

I'm fine with supporting method-chaining here, just some minor stylistic grumbles.

File: example.js (hunk 1 above)
Comment (wayneparrott) [reply example.js]: +1

File: readme.md (hunk 2 above)
Comment (wayneparrott) [reply readme.md]: I revised it to "Returns w." Push back as needed. thx.

**Comment by derhuerst** [comment]
Published as `live-moving-average@1.1.0`. 🎉

**Pull request merged**